### DISCUSSION TITLE
How to reference the same table twice with 2 fields.

### DISCUSSION THREAD

**Discussion by kennarddh** [prisma/prisma#25547] [Q&A]
### Question

I want Articles to have 2 reference to ArticlesImages. One for images array and one for the thumbnail.

`images` field is one to many and `thumbnailImage` field is one to one.

But this error on the line 30 `images ArticlesImages[]`.

> Error validating model "Articles": Ambiguous relation detected. The fields `images` and `thumbnailImage` in model `Articles` both refer to `ArticlesImages`. Please provide different relation names for them by adding `@relation(<name>).

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
generator client {
    provider        = "prisma-client-js"
    binaryTargets   = ["native", "linux-musl-openssl-3.0.x"]
}

datasource db {
    provider = "mysql"
    url      = env("DATABASE_URL")
}

model ArticlesImages {
    id Int @id @default(autoincrement())

    fileName String @db.VarChar(256)

    article   Articles @relation(fields: [articleID], references: [id])
    articleID Int

    createdAt DateTime   @default(now())
    updatedAt DateTime   @updatedAt
}

model Articles {
    id Int @id @default(autoincrement())

    images ArticlesImages[]
    thumbnailImage ArticlesImages
}
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: Windows
- Database: MariaDB
- Node.js version: 22.10.0

- Run `prisma -v` to see your Prisma version and paste it
Environment variables loaded from .env
prisma : 5.18.0
@prisma/client : 5.18.0
Computed binaryTarget : windows
Operating System : win32
Architecture : x64
Node.js : v20.12.2
Query Engine (Node-API) : libquery-engine 4c784e32044a8a016d99474bd02a3b6123742169 (at node_modules@prisma\engines\query_engine-windows.dll.node)
Schema Engine : schema-engine-cli 4c784e32044a8a016d99474bd02a3b6123742169 (at node_modules@prisma\engines\schema-engine-windows.exe)
Schema Wasm : @prisma/prisma-schema-wasm 5.18.0-25.4c784e32044a8a016d99474bd02a3b6123742169
Default Engines Hash : 4c784e32044a8a016d99474bd02a3b6123742169
Studio : 0.502.0
Preview Features : fullTextIndex, fullTextSearch

**Comment by ludralph** [comment]
Hi @kennarddh 👋 

To define two distinct relationships between `Articles` and `ArticlesImages` — one for an array of images and another for a single thumbnail image — use unique `@relation` names and an additional foreign key field. In the corrected schema:

The `images` field in `Articles` uses `@relation("ArticleImages")` to establish a one-to-many relationship with `ArticlesImages`, connecting an article with multiple images.

The `thumbnailImage` field in `Articles` uses `@relation("ArticleThumbnail")` and the additional foreign key field `articleThumnailID` in `ArticlesImages` to create a one-to-one relationship for a single thumbnail image.


```prisma

model ArticlesImages {
  id       Int    @id @default(autoincrement())
  fileName String @db.VarChar(256)

  article   Articles @relation("ArticleImages", fields: [articleID], references: [id])
  articleID Int

  // Optional relation for the thumbnail image
  thumbnailArticle  Articles? @relation("ArticleThumbnail", fields: [articleThumnailID], references: [id])
  articleThumnailID Int?      @unique

  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}

model Articles {
  id Int @id @default(autoincrement())

  // One-to-many relationship with ArticlesImages for the array of images
  images ArticlesImages[] @relation("ArticleImages")

  // One-to-one relationship with ArticlesImages for the thumbnail
  thumbnailImage ArticlesImages? @relation("ArticleThumbnail")
}
```
See the [documentation](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations#disambiguating-relations) on disambuigating relations. Let me know if this helps.

**Reply by kennarddh** [reply]
But this way, the `thumbnailImage` is optional. How to make it always required.

**Reply by ludralph** [reply]
Hi @kennarddh 

To make it required, we can modify the schema slightly
```prisma
model ArticlesImages {
  id        Int      @id @default(autoincrement())
  fileName  String   @db.VarChar(256)
  article   Articles? @relation("ArticleImages", fields: [articleID], references: [id])
  articleID Int?
  thumbnailFor Articles? @relation("ThumbnailImage")
  createdAt DateTime @default(now())
  updatedAt DateTime @updatedAt
}

model Articles {
  id             Int              @id @default(autoincrement())
  images         ArticlesImages[] @relation("ArticleImages")
  thumbnailImage ArticlesImages   @relation("ThumbnailImage", fields: [thumbnailImageId], references: [id])
  thumbnailImageId Int
}
```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
